include/transform_output/transform_output.hpp

```diff
@@ -0,0 +1,212 @@
+ *  file you want to transform and the name of the output file and generates 
+ *  the new output file.
+ *  
+ *  This function can generate formatted output for the following components
+ *      1- ABP Simulator
+ *      2- Sender
+ *      3- Receiver
+ *      4- Subnet
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+
+using namespace std;
+
+/**
+ * This function is used to trim spaces present on the right side of a string.
+ * It takes two parameters as below.
+ * @param str of type string
+ * @param chars of type string
+ */
+std::string& rtrim(std::string& str, const std::string& chars = "\t\n\v\f\r ")
+{
+    str.erase(str.find_last_not_of(chars) + 1);
+    return str;
+}
+
+void transform_output(char *ip_file_name, char *op_file_name) {
+	
+	/**
+	 * Creating a file pointer which will hold input file name
+	 */
+	FILE *fp;
+
+	/**
+	 * temp array is used to store the one line read from input file.
+	 */
+	char line[512],*pattern="00:";
+	
+	/**
+	 * string variables to hold imtermediate sub-strings extracted from input file
+	 */
+	std::string in, time, port, port_value, component;
+	
+	/**
+	 * ofstream creates a new output file if not present or clean the 
+	 * existing one before writing data to it.
+	 */
+	ofstream outputfile(op_file_name);
+
+	/**
+	 * setting the column names for our output file.
+	 * setw() function is used to give indentation in the output file.
+	 */
+	outputfile <<setw(7)<<"Time"<<setw(20)<<"Value"<<setw(14)<<"Port"<<setw(22)<<"Component"<<endl;
+	outputfile <<"----------------------------------------------------------------------"<<endl;
+	
+	//gcc users
+	if((fp = fopen(ip_file_name, "r")) == NULL) {
+		printf("can't open %s \n",ip_file_name);
+	}
+
+	/**
+	 * fgets loop iterate through the input file line by line until EOF and 
+	 * store the line read in temp[] char array.
+	 */
+	while(fgets(line, 512, fp) != NULL) {
+
+		/**
+		 * The strstr() function searches the given string in the specified main string and 
+		 * returns the pointer to the first occurrence of the given string.
+		 * temp holds line read from file. str contains a predefined pattern to search through input file.
+		 * if loop specifically evaluates that we are searching for only time in input file. If the pattern 'str'
+		 * matches a time value in input file then if loop will be executed and we store the  time value in 'time'
+		 * string variable. We perform rtrim() on temp to remove any spaces present after string.
+		 * else loop will be executed if the line read from input file doesn't contain a time value.
+		 * for each time value read by 'if loop', we retain it and read lines until we encounter another time value.
+		 * This is done beacause each 'time' value will have port, port value and component  associated with it.
+		 */
+
+		if((strstr(line, pattern)) != NULL) {
+			time = line;
+			time = rtrim(time);
+		}else{

```

Comment:
line 96 else should be in next line
check for other else if and else also

Reply:
No else must start after ending flower brackets of if loop. See coding conventions established by professor. We confirmed with here many times and she said to keep else immediately after if }

Reply:
ok then follow what professor said